test/all-commands.spec.ts

```diff
@@ -0,0 +1,163 @@
+import { Faker, faker } from '@faker-js/faker';
+import { exec } from 'node:child_process';
+import { promisify } from 'node:util';
+import { describe, expect, it } from 'vitest';
+
+type FakerMap = {
+  [moduleKey in keyof Faker]?: ReadonlyArray<keyof Faker[moduleKey] | '*'>;
+};
+
+type LiteralUnion<TSuggested extends string> =
+  | TSuggested
+  | (string & { zz_IGNORE_ME?: never });
+
+const execPromise = promisify(exec);
+
+const DEPRECATED_MAP: {
+  [module: string]: ReadonlyArray<string>;
+} = {
+  company: [
+    'bs',
+    'bsAdjective',
+    'bsBuzz',
+    'bsNoun',
+    'companySuffix',
+    'suffixes',
+  ],
+  datatype: [
+    'array',
+    'bigInt',
+    'datetime',
+    'float',
+    'hexadecimal',
+    'json',
+    'number',
+    'string',
+    'uuid',
+  ],
+  finance: ['account', 'mask'],
+  git: ['shortSha'],
+  image: [
+    'abstract',
+    'animals',
+    'business',
+    'cats',
+    'city',
+    'fashion',
+    'food',
+    'image',
+    'imageUrl',
+    'nature',
+    'nightlife',
+    'people',
+    'sports',
+    'technics',
+    'transport',
+  ],
+  location: ['stateAbbr', 'streetName', 'cityName', 'zipCodeByState'],
+  random: ['*'],
+} satisfies FakerMap;
+
+const REQUIRE_PARAMS_MAP: {
+  [module: string]: ReadonlyArray<string>;
+} = {
+  date: ['between', 'betweens'],
+  helpers: ['*'],
+  string: ['fromCharacters'],
+} satisfies FakerMap;
+
+/**
+ * Checks whether a Faker module is valid.
+ * Invalid modules are for example private properties.
+ */
+function isValidModule(moduleKey: string): boolean {
+  const IGNORE_MODULES: LiteralUnion<keyof Faker>[] = [
+    '_randomizer',
+    'definitions',
+    'rawDefinitions',
+  ];
+
+  return !(
+    IGNORE_MODULES.includes(moduleKey) ||
+    DEPRECATED_MAP[moduleKey]?.includes('*') ||
+    REQUIRE_PARAMS_MAP[moduleKey]?.includes('*')
+  );
+}
+
+/**
+ * Checks whether a Faker method was removed, but still exists on the module.
+ * These methods usually exist for JS backwards-compatibility.
+ */
+function isRemoved(moduleKey: string, entryKey: string): boolean {
+  const REMOVED_METHODS: {
+    [module: string]: ReadonlyArray<string>;
+  } = {
+    random: ['locale'],
+  };
+
+  const moduleRef = REMOVED_METHODS[moduleKey] ?? [];
+  return moduleRef.includes(entryKey);
```

Comment:
Note that your code will not result in the same value if `moduleKey` does not exist on `REMOVED_METHODS`. It will return `undefined` instead of `false`.

Reply:
Then add `?? false`